site/content/how-to/traffic-management/snippets.md

```diff
@@ -0,0 +1,362 @@
+
+```yaml
+kubectl apply -f - <<EOF
+apiVersion: gateway.nginx.org/v1alpha1
+kind: SnippetsFilter
+metadata:
+  name: tea-rate-limiting-sf
+spec:
+  snippets:
+    - context: http
+      value: limit_req_zone $binary_remote_addr zone=teazone:10m rate=1r/s;
+    - context: http.server.location
+      value: limit_req zone=teazone burst=3;
+EOF
+```
+
+This `SnippetsFilter` is the same as the one applied to the coffee HTTPRoute, however it removes the `nodelay` setting
+on the `limit_req` directive. This forces a delay on the incoming requests to match the rate set in `limit_req_zone`.
+
+Verify that the `SnippetsFilter` is Accepted:
+
+```shell
+kubectl describe snippetsfilters.gateway.nginx.org tea-rate-limiting-sf
+```
+
+You should see the following status:
+
+```text
+Status:
+  Controllers:
+    Conditions:
+      Last Transition Time:  2024-10-21T22:20:24Z
+      Message:               SnippetsFilter is accepted
+      Observed Generation:   1
+      Reason:                Accepted
+      Status:                True
+      Type:                  Accepted
+    Controller Name:         gateway.nginx.org/nginx-gateway-controller
+Events:                      <none>
+```
+
+Verify that the tea `HTTPRoute` which had an `InvalidFilter` condition earlier, no longer has that condition.
+
+```shell
+kubectl describe httproutes.gateway.networking.k8s.io tea
+```
+
+You should see the following conditions:
+
+```text
+Conditions:
+      Last Transition Time:  2024-10-23T00:33:08Z
+      Message:               The route is accepted
+      Observed Generation:   2
+      Reason:                Accepted
+      Status:                True
+      Type:                  Accepted
+      Last Transition Time:  2024-10-23T00:33:08Z
+      Message:               All references are resolved
+      Observed Generation:   2
+      Reason:                ResolvedRefs
+      Status:                True
+      Type:                  ResolvedRefs
+```
+
+Test that the `SnippetsFilter` is configured and has successfully applied the rate limiting NGINX configuration changes.
+
+Send a request to tea:
+
+```shell
+curl --resolve cafe.example.com:$GW_PORT:$GW_IP http://cafe.example.com:$GW_PORT/tea
+```
+
+This request should receive a response from the tea Pod:
+
+```text
+Server address: 10.244.0.7:8080
+Server name: tea-76c7c85bbd-cf8nz
+```
+
+When processing a single request, the rate limiting configuration has no noticeable effect. Try sending
+multiple requests.
+
+```shell
+for i in `seq 1 10`; do curl --resolve cafe.example.com:$GW_PORT:$GW_IP http://cafe.example.com:$GW_PORT/tea; done
+```
+
+You should see all successful responses from the tea Pod, but they should be spaced apart roughly one second each as
+expected through the rate limiting configuration.
+
+You've now used the `SnippetFilter` resource to configure two distinct rate limiting rules to different backend applications through the use of Snippets.
```

Comment:
```suggestion
You've now used the `SnippetsFilter` resource to configure two distinct rate limiting rules to different backend applications through the use of Snippets.
```

Reply:
You can probably drop "now" from the sentence, too.

Reply:
Going to keep the now if that's alright with you, seems a little weird without it and I think in transitions into the following statement: "For an alternative method..." a bit better

Reply:
I made an edit suggestion for alternative phrasing following the preceding sentence.